pensieve/pensieve.py

```diff
@@ -245,99 +173,165 @@ def __init__(self, text, par_id=None, doc=None):
-        self.id = par_id
-        self.spacy_doc = NLP(text)
         self.words = self.build_words_dict()
+        self.id = par_id
 
-    def build_words_dict(self):
+    def extract_times(self):
         """
-        Use textacy to extract entities and main verbs from the paragraph.
+        Extract times mentioned in the text and add book time information
         """
-        words_dict = {'times': Counter(),
-                      'verbs': Counter(),
-                      'names': Counter(),
-                      'places': Counter(),
-                      'things': Counter(),
-                      'mood': None,
-                      'img_url': None}
-        times = textacy.extract.named_entities(self.spacy_doc, include_types=['DATE', 'TIME', 'EVENT'])
-        main_verbs = textacy.spacy_utils.get_main_verbs_of_sent(self.spacy_doc)
-        names = textacy.extract.named_entities(self.spacy_doc, include_types=['PERSON'])
-        places = textacy.extract.named_entities(self.spacy_doc, include_types=['LOC', 'GPE', 'FACILITY'])
-        things = textacy.extract.named_entities(self.spacy_doc, include_types=['ORG', 'NORP', 'WORK_OF_ART', 'PRODUCT'])
-        for time in times:
-            words_dict['times'][time.text] += 1
-        for verb in main_verbs:
-            words_dict['verbs'][verb.text] += 1
-        for name in names:
-            words_dict['names'][name.text] += 1
-        for place in places:
-            words_dict['places'][place.text] += 1
-        for thing in things:
-            words_dict['things'][thing.text] += 1
-        return words_dict
-
-    def sanitize_places_and_things(self, freq_cut=100):
+        times = []
+        include_types = ['DATE', 'TIME', 'EVENT']
+        for time in textacy.extract.named_entities(self.spacy_doc,
+                                                   include_types=include_types):
+            times.append(time)
+        return times
+
+    def extract_verbs(self, n_verbs=3):
         """
-        Get a sanitized version of paragraph places and things.
-        Get places that don't appear in the most common doc names, and
-        things that don't appear in the most common doc places.
-
-        Args:
-            freq_cut: frequency to define "commonness" within the doc
+        Extract verbs in the paragraph
+        """
+        verb_ranking = [(x, x.rank) for x in textacy.spacy_utils.get_main_verbs_of_sent(self.spacy_doc)]
+        verbs = []
+        for verb in sorted(verb_ranking, key=lambda x: -x[1])[:n_verbs]:
+            text = verb[0].lemma_
+            if text[0] in string.ascii_lowercase:
+                verbs.append(text)
+        return verbs
+
+    def extract_names(self):
+        """
+        Extract names in paragraph
+        """
+        names = []
+        for name in textacy.extract.named_entities(self.spacy_doc,
+                                                   include_types=['PERSON']):
+            if name.text[-2] not in string.ascii_lowercase:
```

Comment:
There is an instance in book 4 where '...' (as a single character) is flagged as a person and breaks the code on this line